tests/test_translation_plugin.py

```diff
@@ -1,170 +1,90 @@
-    assert isinstance(op, Initialize)
-    assert np.array_equal(op.params, state)
-
-
-def test_complex_plus_with_global_phase() -> None:
-    c = QuantumCircuit(2, 1)
-    state = np.array([1, 1]) / np.sqrt(2) * np.exp(3j)
-    c.initialize(state, 0)
-    new_c = _pm.run(c)
-    instructions = new_c.get_instructions('initialize')
-    assert len(instructions) == 1
-    assert c.find_bit(instructions[0].qubits[0]).index == 0
-    op = instructions[0].operation
-    assert isinstance(op, Initialize)
-    assert np.array_equal(op.params, state)
-
-
-def test_complex_minus() -> None:
-    c = QuantumCircuit(2, 1)
-    state = np.array([1, -1]) / np.sqrt(2)
-    c.initialize(state, 0)
-    new_c = _pm.run(c)
-    instructions = new_c.get_instructions('initialize')
-    assert len(instructions) == 1
-    assert c.find_bit(instructions[0].qubits[0]).index == 0
-    op = instructions[0].operation
-    assert isinstance(op, Initialize)
-    assert np.array_equal(op.params, state)
-
-
-def test_complex_0() -> None:
-    c = QuantumCircuit(2, 1)
-    c.initialize([1, 0], 0)
-    new_c = _pm.run(c)
-    instructions = new_c.get_instructions('reset')
-    assert len(instructions) == 1
-    assert c.find_bit(instructions[0].qubits[0]).index == 0
-    op = instructions[0].operation
-    assert isinstance(op, Reset)
-
-
-def test_complex_1() -> None:
-    c = QuantumCircuit(2, 1)
-    c.initialize([0, 1], 0)
-    new_c = _pm.run(c)
-    instructions = new_c.get_instructions('initialize')
-    assert len(instructions) == 1
-    assert c.find_bit(instructions[0].qubits[0]).index == 0
-    op = instructions[0].operation
-    assert isinstance(op, Initialize)
-    assert np.array_equal(op.params, [0, 1])
-
-
-def test_complex_unsupported() -> None:
-    c = QuantumCircuit(2, 1)
-    state = np.array([1, -3j])
-    state /= np.sqrt(np.sum(state * np.conj(state)))
-    c.initialize(state, 0)
-    with pytest.raises(AliceBobTranspilationException):
-        _pm.run(c)
+
+def _check_initialize(circuit: QuantumCircuit, expected: str) -> None:
+    initializes = circuit.get_instructions('initialize')
+    assert len(initializes) == 1
+    assert isinstance(initializes[0].operation, Initialize)
+    params = initializes[0].operation.params
+    assert params == list(expected)
+
+
+def test_int_to_label() -> None:
+    pm = PassManager([IntToLabelInitializePass()])
+
+    # int(2) should map to '10'
+    circ = QuantumCircuit(2)
+    circ.initialize(2)
+    transpiled = pm.run(circ)
+    _check_initialize(transpiled, '10')
+
+    # a string label should pass unchanged
+    circ = QuantumCircuit(2)
+    circ.initialize('+0')
+    transpiled = pm.run(circ)
+    _check_initialize(transpiled, '+0')
+
+    # a state vector should fail
+    circ = QuantumCircuit(2)
+    circ.initialize([1, 0, 0, 0])
+    with pytest.raises(TranspilerError):
+        pm.run(circ)
+
+
+def _check_initializes(circuit: QuantumCircuit, expected: List[str]) -> None:
```

Comment:
nit: wdyt of moving this function at beginning of the file, just after _check_initialize?

Reply:
Done!